thavalon-webapp/src/components/Navbar.tsx

```diff
@@ -0,0 +1,154 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import React, {useState} from 'react';
+import {NavLink} from 'react-router-dom';
+import styled from 'styled-components';
+import {faBars} from '@fortawesome/free-solid-svg-icons';
+
+function openMenu(event: any) {
+    let x = document.getElementById("navbar");
+    if (x === null) {
+        return;
+    }
+    if (x.classList.contains("responsive")) {
+        x.classList.remove("responsive");
+    } else if (x.classList.contains("topnav")) {
+        x.classList.add("responsive");
+    }
+
+    event.preventDefault();
+}
+
+type NavbarProps = {
+    loggedIn: boolean
+};
+
+function Navbar(props: NavbarProps) {    
+    return (
+        <NavbarContainer id="navbar" className="topnav">
+            <NavbarItemLeft exact to="/" activeClassName="active" id="homeLink">Home</NavbarItemLeft>
+            <NavbarItemLeft to="/rules" activeClassName="active">Rules</NavbarItemLeft>
+            {!props.loggedIn &&
+                <NavbarItemRight to="/login" activeClassName="active">Log In</NavbarItemRight>
+            }
+            <NavbarItemRight to="" className="icon" onClick={openMenu}>
+                    <FontAwesomeIcon icon={faBars} />
+            </NavbarItemRight>
+            {props.loggedIn &&
+                <RightSpan id="rightContainer">
+                    <SubNavbarItemLeft to="/stats" className="nav-item" activeClassName="active">Stats</SubNavbarItemLeft>
+                    <SubNavbarItemLeft to="/account" className="nav-item" activeClassName="active">Account</SubNavbarItemLeft>
+                    <SubNavbarItemLeft to="/logout" className="nav-item" activeClassName="active">Log Out</SubNavbarItemLeft>
+                </RightSpan>
+            }
+        </NavbarContainer>
+    )
+}
+
+export default Navbar;
+
+const RightSpan = styled.span`
+    display: inline-block;
+    float: right;
+
+    @media screen and (max-width: 600px) {
+        float: none;
+        width: 100%;
+    }
+`;
+
+const NavbarContainer = styled.div`
+    background-color: #333;
+    overflow: hidden;
+
+    @media screen and (max-width: 600px) {
+        &.responsive {
+            a, #rightContainer a  {
+                &.icon {
+                    position: absolute;
+                    right: 0;
+                    top: 0;
+                }
+
+                float: none;
+                display: block;
+                text-align: left;
+            }
+        }
+    }
+
+    .icon {
+        display: none;
+    }
+`;
+
+const NavbarItem = styled(NavLink)`

```

Comment:
Should use scss instead of styled-components

Reply:
Not my code.